### ISSUE TITLE
calculate the camouflaged object edge

### ISSUE THREAD

**Issue by znshi** [thograce/BGNet#4]
Thanks for sharing the code. 
I don't know how you calculate the camouflaged object edge( ground truth: Ge) based on the camouflaged object mask. Because the edge ground truth will be used in the loss function.
So, can you tell me how to get the camouflaged object edge map?

**Comment by thograce** [comment]
The camouflaged object edge maps are provided in the training dataset.

**Comment by thograce** [comment]
> Thanks for sharing the code. I don't know how you calculate the camouflaged object edge( ground truth: Ge) based on the camouflaged object mask. Because the edge ground truth will be used in the loss function. So, can you tell me how to get the camouflaged object edge map?

The camouflaged object edge maps are provided in the training dataset.

**Comment by znshi** [comment]
> 
Thank you very much for your reply, I saw the edge maps you provided in the training dataset. Since I am doing the direction of medical image segmentation, I would like to see the experimental results in our direction of the image dataset, and I sincerely hope to share the code of the computer edge maps. My email: shizn17@jlu.edu.cn

**Comment by znshi** [comment]
> > Thanks for sharing the code. I don't know how you calculate the camouflaged object edge( ground truth: Ge) based on the camouflaged object mask. Because the edge ground truth will be used in the loss function. So, can you tell me how to get the camouflaged object edge map?
> 
> The camouflaged object edge maps are provided in the training dataset.

sorry，sorry，上一个回复的邮件写错了，shizn17@mails.jlu.edu.cn,真心期待你的回复，谢谢谢谢。

**Comment by znshi** [comment]
> > Thanks for sharing the code. I don't know how you calculate the camouflaged object edge( ground truth: Ge) based on the camouflaged object mask. Because the edge ground truth will be used in the loss function. So, can you tell me how to get the camouflaged object edge map?
> 
> The camouflaged object edge maps are provided in the training dataset.

sorry，sorry，上一个回复的邮件写错了，shizn17@mails.jlu.edu.cn,真心期待你的回复，谢谢谢谢。

**Comment by thograce** [comment]
> > > Thanks for sharing the code. I don't know how you calculate the camouflaged object edge( ground truth: Ge) based on the camouflaged object mask. Because the edge ground truth will be used in the loss function. So, can you tell me how to get the camouflaged object edge map?
> > 
> > 
> > The camouflaged object edge maps are provided in the training dataset.
> 
> sorry，sorry，上一个回复的邮件写错了，[shizn17@mails.jlu.edu.cn](mailto:shizn17@mails.jlu.edu.cn),真心期待你的回复，谢谢谢谢。

I am not the builder of the training dataset. They are from [SINet](https://github.com/DengPingFan/SINet).
But I can provide you with the code for my personal use to generate the edge maps based on the masks:

```
import numpy as np
import cv2

img = cv2.imread('img.png', cv2.IMREAD_GRAYSCALE)
img = np.array(img,dtype= np.float64)
g_x = np.array([[-1, 0, 1], [-1, 0, 1], [-1, 0, 1]])
g_y = np.array([[-1, -1, -1], [0, 0, 0], [1, 1, 1]])

img_g_x = cv2.filter2D(img, -1, g_x)
img_g_y = cv2.filter2D(img, -1, g_y)
img_graid = np.sqrt(img_g_x ** 2 + img_g_y ** 2)

cv2.imwrite('e.png', img_graid)
```

**Comment by znshi** [comment]
> > > > Thanks for sharing the code. I don't know how you calculate the camouflaged object edge( ground truth: Ge) based on the camouflaged object mask. Because the edge ground truth will be used in the loss function. So, can you tell me how to get the camouflaged object edge map?
> > > 
> > > 
> > > The camouflaged object edge maps are provided in the training dataset.
> > 
> > 
> > sorry，sorry，上一个回复的邮件写错了，[shizn17@mails.jlu.edu.cn](mailto:shizn17@mails.jlu.edu.cn),真心期待你的回复，谢谢谢谢。
> 
> I am not the builder of the training dataset. They are from [SINet](https://github.com/DengPingFan/SINet). But I can provide you with the code for my personal use to generate the edge maps based on the masks:
> 
> ```
> import numpy as np
> import cv2
> 
> img = cv2.imread('img.png', cv2.IMREAD_GRAYSCALE)
> img = np.array(img,dtype= np.float64)
> g_x = np.array([[-1, 0, 1], [-1, 0, 1], [-1, 0, 1]])
> g_y = np.array([[-1, -1, -1], [0, 0, 0], [1, 1, 1]])
> 
> img_g_x = cv2.filter2D(img, -1, g_x)
> img_g_y = cv2.filter2D(img, -1, g_y)
> img_graid = np.sqrt(img_g_x ** 2 + img_g_y ** 2)
> 
> cv2.imwrite('e.png', img_graid)
> ```

收到收到，谢谢谢谢~祝paper多多，

**Comment by thograce** [comment]
> > > > > Thanks for sharing the code. I don't know how you calculate the camouflaged object edge( ground truth: Ge) based on the camouflaged object mask. Because the edge ground truth will be used in the loss function. So, can you tell me how to get the camouflaged object edge map?
> > > > 
> > > > 
> > > > The camouflaged object edge maps are provided in the training dataset.
> > > 
> > > 
> > > sorry，sorry，上一个回复的邮件写错了，[shizn17@mails.jlu.edu.cn](mailto:shizn17@mails.jlu.edu.cn),真心期待你的回复，谢谢谢谢。
> > 
> > 
> > I am not the builder of the training dataset. They are from [SINet](https://github.com/DengPingFan/SINet). But I can provide you with the code for my personal use to generate the edge maps based on the masks:
> > ```
> > import numpy as np
> > import cv2
> > 
> > img = cv2.imread('img.png', cv2.IMREAD_GRAYSCALE)
> > img = np.array(img,dtype= np.float64)
> > g_x = np.array([[-1, 0, 1], [-1, 0, 1], [-1, 0, 1]])
> > g_y = np.array([[-1, -1, -1], [0, 0, 0], [1, 1, 1]])
> > 
> > img_g_x = cv2.filter2D(img, -1, g_x)
> > img_g_y = cv2.filter2D(img, -1, g_y)
> > img_graid = np.sqrt(img_g_x ** 2 + img_g_y ** 2)
> > 
> > cv2.imwrite('e.png', img_graid)
> > ```
> 
> 收到收到，谢谢谢谢~祝paper多多，

我毕业了，不用发论文了，哈哈哈哈！祝你科研顺利~

**Comment by znshi** [comment]
> > > > > > Thanks for sharing the code. I don't know how you calculate the camouflaged object edge( ground truth: Ge) based on the camouflaged object mask. Because the edge ground truth will be used in the loss function. So, can you tell me how to get the camouflaged object edge map?
> > > > > 
> > > > > 
> > > > > The camouflaged object edge maps are provided in the training dataset.
> > > > 
> > > > 
> > > > sorry，sorry，上一个回复的邮件写错了，[shizn17@mails.jlu.edu.cn](mailto:shizn17@mails.jlu.edu.cn),真心期待你的回复，谢谢谢谢。
> > > 
> > > 
> > > I am not the builder of the training dataset. They are from [SINet](https://github.com/DengPingFan/SINet). But I can provide you with the code for my personal use to generate the edge maps based on the masks:
> > > ```
> > > import numpy as np
> > > import cv2
> > > 
> > > img = cv2.imread('img.png', cv2.IMREAD_GRAYSCALE)
> > > img = np.array(img,dtype= np.float64)
> > > g_x = np.array([[-1, 0, 1], [-1, 0, 1], [-1, 0, 1]])
> > > g_y = np.array([[-1, -1, -1], [0, 0, 0], [1, 1, 1]])
> > > 
> > > img_g_x = cv2.filter2D(img, -1, g_x)
> > > img_g_y = cv2.filter2D(img, -1, g_y)
> > > img_graid = np.sqrt(img_g_x ** 2 + img_g_y ** 2)
> > > 
> > > cv2.imwrite('e.png', img_graid)
> > > ```
> > 
> > 
> > 收到收到，谢谢谢谢~祝paper多多，
> 
> 我毕业了，不用发论文了，哈哈哈哈！祝你科研顺利~

哎，慕了~~那工作生活顺利哈~

**Comment by thograce** [comment]
[https://github.com/JXingZhao/EGNet/blob/master/sal2edge.m](https://github.com/JXingZhao/EGNet/blob/master/sal2edge.m)
